sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -68,20 +69,23 @@
 _LOGGER = logging.getLogger(__name__)
 _Address = collections.namedtuple("_Address", "hostname path")
 
+def _is_local_endpoint(endpoint: str) -> bool:
+    return re.match("^(127\.[\d.]+|[0:]+1|localhost)", endpoint.lower()) is not None # pylint:disable=anomalous-backslash-in-string
```

Comment:
```suggestion
    return re.match("^(127\\.[\\d.]+|[0:]+1|localhost)", endpoint.lower()) is not None # pylint:disable=anomalous-backslash-in-string
```
The backslashes need to be escaped. This might remove the need of the pylint disable